### PR TITLE
fix(www): responsive layout for `/guidelines/color` + convert to theme-ui

### PR DESCRIPTION
**Pull request by abhishekjakhar**
## Description

1. Converted the guidelines/color page to theme-ui.
2. Fixed the responsive layout.
3. Fixed the color modal on responsive.

Local URL: http://localhost:8000/guidelines/color/
Production URL: https://www.gatsbyjs.org/guidelines/color/

##### iPhone(x)

iPhone(x)-Actual

iPhone(X)-Modal-Actual

##### iPad

iPad-Actual

##### iPhone(x)

iPhone(X)-Fixed

iPhone(X)-Modal-Fixed

##### iPad

iPad-Fixed

iPad-Modal-Fixed

### CODE REVIEW

File: www/src/pages/guidelines/color.js [hunk 1]
```diff
@@ -25,15 +28,17 @@ import ColorModal from "../../components/guidelines/color/modal"
 
 import palette from "../../utils/guidelines/extend-palette-info"
 
-const LegacyColorIcon = ({ textColor }) => (
```
Comment (fk) [inline www/src/pages/guidelines/color.js:28]: 🤦 🙄 LOL wondering what I was thinking there with `textColor` ;) — really appreciate your attention to these details! <3

Comment (abhishekjakhar) [reply www/src/pages/guidelines/color.js:28]: I can understand, I also added `box-sizing: border-box` to div even though it was declared using `*` selector :see_no_evil:
hahahahah

File: www/src/components/guidelines/color/modal.js [hunk 2]
```diff
@@ -1,57 +1,66 @@
...
+      cursor: `pointer`,
+      outline: 0,
+      WebkitAppearance: `none`,
+      pb: 6,
+      transition: t =>
+        `all ${t.transition.speed.fast} ${t.transition.curve.default}`,
       ":hover, :focus": {
         ...focusStyle,
-        color: themeGet(`colors.orange.50`)(props),
+        color: t => t.colors.orange[50],
```
Comment (fk) [inline www/src/components/guidelines/color/modal.js:56]:
```suggestion
        color: `orange.50`,
```

No need to grab this via `t`/[functional values](https://theme-ui.com/sx-prop/#functional-values) here as the CSS `color` prop is mapped to the `theme-ui` `colors` scale directly.

File: www/src/components/guidelines/color/modal.js [hunk 3]
```diff
@@ -138,7 +151,16 @@ const modalContent = (palette, color) => {
       const textColor = getTextColor(c.contrast)
 
       colors.push(
-        <Box width="100%" display={{ lg: `flex` }} key={`palette-${index}`}>
+        <Box
+          sx={{
+            display: `block`,
```
Comment (fk) [inline www/src/components/guidelines/color/modal.js:156]: Not sure, but do we need the explicit `display: block` for `Box` (https://theme-ui.com/components/box makes it look like this renders as a `div` by default, which has `display: block`)?

File: www/src/components/guidelines/color/modal.js [hunk 4]
```diff
@@ -178,7 +205,13 @@ const modalContent = (palette, color) => {
               {c.contrast.colorOnWhite.toFixed(2)} /{` `}
               {getA11yLabel(c.a11y)}
             </Text>
-            <Box width="100%" flexShrink={0} css={{ "& p": { color: c.hex } }}>
+            <Box
+              sx={{
+                width: `100%`,
+                flexShrink: 0,
+              }}
+              css={{ "& p": { color: c.hex } }}
```
Comment (fk) [inline www/src/components/guidelines/color/modal.js:213]: I think these should be fine to drop in the `sx` object — nesting should work the same as it does for the `css` prop.

File: www/src/components/guidelines/color/modal.js [hunk 5]
```diff
@@ -196,19 +229,34 @@ const ColorModal = ({ palette, color, handleModalClose }) => {
 
   return (
     <>
-      <Flex alignItems="baseline" p={{ xxs: 6, md: 8 }}>
+      <Flex
+        sx={{
+          alignItems: `baseline`,
+          p: 6,
+          [mediaQueries.md]: {
```
Comment (fk) [inline www/src/components/guidelines/color/modal.js:236]: Again, I think these are ideal for `theme-ui`s "responsive array" syntax. You can omit values for breakpoints via `null`, see e. g. https://github.com/gatsbyjs/gatsby/blob/79986bdf4da863dea49d20ec69098f6dc0f56fcd/www/src/components/navigation.js#L160

File: www/src/components/guidelines/color/modal.js [hunk 6]
```diff
@@ -196,19 +229,34 @@ const ColorModal = ({ palette, color, handleModalClose }) => {
...
+          },
+        }}
+      >
         <Heading color="black" mr={4} mt={0}>
           {palette[color].name}
         </Heading>
         <CloseButton
           onClick={handleModalClose}
           aria-label={`Close “${palette[color].name}” modal`}
-          paddingBottom="6"
```
Comment (fk) [inline www/src/components/guidelines/color/modal.js:206]: 👍 🙏

File: www/src/components/guidelines/color/modal.js [hunk 7]
```diff
@@ -196,19 +229,34 @@ const ColorModal = ({ palette, color, handleModalClose }) => {
...
         </CloseButton>
       </Flex>
-      <Box p={{ xxs: 6, md: 8 }} bg={base.hex}>
+      <Box
+        sx={{
+          p: 6,
+          bg: base.hex,
+          [mediaQueries.md]: {
+            p: 8,
+          },
```
Comment (fk) [inline www/src/components/guidelines/color/modal.js:257]: See https://github.com/gatsbyjs/gatsby/pull/24537/files#r431252384

**Comment by fk** [comment]
Left a few comments, but I think the only one that we should fix before merging is https://github.com/gatsbyjs/gatsby/pull/24537#pullrequestreview-419361828. LMK about the other stuff! 🙏

**Comment by abhishekjakhar** [comment]
> Beautiful! Again, thanks for all your work here Abhishek! I really appreciate it! 🙏

@fk Thank you very much for reviewing it, I have noted down all the points which can be improved, I will definitely take care of them.

File: www/src/components/guidelines/color/modal.js (hunk 3 above)
Comment (fk) [inline www/src/components/guidelines/color/modal.js:156]:
```suggestion
```

File: www/src/components/guidelines/color/modal.js [hunk 8]
```diff
@@ -178,7 +204,13 @@ const modalContent = (palette, color) => {
               {c.contrast.colorOnWhite.toFixed(2)} /{` `}
               {getA11yLabel(c.a11y)}
             </Text>
-            <Box width="100%" flexShrink={0} css={{ "& p": { color: c.hex } }}>
+            <Box
+              sx={{
+                width: `100%`,
+                flexShrink: 0,
+              }}
```
Comment (fk) [inline www/src/components/guidelines/color/modal.js:211]:
```suggestion
              sx={{
                width: `100%`,
                flexShrink: 0,
                "& p": { color: c.hex }
              }}
```

File: www/src/components/guidelines/color/modal.js [hunk 9]
```diff
@@ -178,7 +204,13 @@ const modalContent = (palette, color) => {
...
+              css={{ "& p": { color: c.hex } }}
```
Comment (fk) [inline www/src/components/guidelines/color/modal.js:212]:
```suggestion
```

**Comment by fk** [comment]
> I have noted down all the points which can be improved, I am working on them.

I applied all the changes that I think should go in this PR before merging; we can address the "responsive array" syntax stuff later on. Merging as soon as checks turn green! ✌️ 🙏

**Comment by abhishekjakhar** [comment]
> > I have noted down all the points which can be improved, I am working on them.
> 
> I applied all the changes that I think should go in this PR before merging; we can address the "responsive array" syntax stuff later on. Merging as soon as checks turn green! ✌️ 🙏

@fk Thank you very much. I got busy with some other stuff.

File: www/src/components/guidelines/color/modal.js [hunk 10]
```diff
@@ -178,7 +204,13 @@ const modalContent = (palette, color) => {
               {c.contrast.colorOnWhite.toFixed(2)} /{` `}
               {getA11yLabel(c.a11y)}
             </Text>
-            <Box width="100%" flexShrink={0} css={{ "& p": { color: c.hex } }}>
+            <Box
+              sx={{
+                width: `100%`,
+                flexShrink: 0,
+                "& p": { color: c.hex }
```
Comment (fk) [inline www/src/components/guidelines/color/modal.js:211]:
```suggestion
                "& p": { color: c.hex },
```

Thank you, Prettier!

**Comment by fk** [comment]
>  I got busy with some other stuff.

No worries! 🤗

**Pull request merged**